File: 프로그래머스/lv1/42748. K번째수/K번째수.js
function solution(array, commands) {
    let returnArr = [];
    for (let idx = 0; idx < commands.length; idx++) {
        let i = commands[idx][0];
        let j = commands[idx][1];
        let k = commands[idx][2];
        
        let answer = [...array].slice(i-1, j).sort((a,b) => (a-b));
        returnArr.push(answer[k-1])
    }
    return returnArr;
}

File: 프로그래머스/1/42748. K번째수/K번째수.py
"""
i번째 숫자부터 j번째 숫자까지 자르고 정렬했을때,
k번째에 있는 수를 구하려고 한다.

TC: O(n log n)
SC: O(n)
"""

def solution(array, commands):
    return [sorted(array[i-1:j])[k-1] for i, j, k in commands]